### PR TITLE
docs: Improve code comments on transactions

### PR DESCRIPTION
**Pull request by philipphofmann**
Improve code comments on methods for capturing transactions.

### CODE REVIEW

File: Sources/Sentry/Public/SentryHub.h [hunk 1]
```diff
@@ -45,27 +45,29 @@ SENTRY_NO_INIT
...
+ * @param name The transaction name.
+ *
+ * @return The created transaction.
  */
 - (SentryTransaction *)startTransactionWithName:(NSString *)name
     NS_SWIFT_NAME(startTransaction(name:));
 
 /**
- * Creates a Transaction bound to the current hub and returns the instance.
+ * Creates a transaction bound to the and returns the instance.
```
Comment (marandaneto) [inline Sources/Sentry/Public/SentryHub.h:66]:
```suggestion
 * Creates a transaction bound to the hub and returns the instance.
```

File: Sources/Sentry/Public/SentryHub.h [hunk 2]
```diff
@@ -45,27 +45,29 @@ SENTRY_NO_INIT
 /**
  * Captures a transaction and sends it to Sentry.
  *
- * @param transaction The Transaction to send to Sentry.
+ * @param transaction The transaction to send to Sentry.
```
Comment (marandaneto) [inline Sources/Sentry/Public/SentryHub.h:48]: L: our official docs `sentry-docs` always `PascalCase` names that are meant as features, which is the case of a `Transaction`, represents a feature and not merely the wording transaction itself.
wondering why would we do differently.

eg https://docs.sentry.io/product/releases/health/#crash
you can see the wording `Discover` and `Issues` using Pascal case.

I'm not against and fine with the changes, just wanted to understand the reasoning

Comment (philipphofmann) [reply Sources/Sentry/Public/SentryHub.h:48]: Here, I think we are not referring to the feature but instead to the class. So I think "transaction" is appropriate. @brustolin, the discussion here could be interesting for you as well.

Comment (brustolin) [reply Sources/Sentry/Public/SentryHub.h:48]: Understood

**Review by marandaneto — APPROVED** [review]

**Pull request merged**